### PR TITLE
#502 GzipInput and GzipOutput

### LINKED ISSUE #502: GzipInput and GzipOutput
Let's implement a way to encode and decode `Bytes` using `Gzip` compress algorithm. I'll handle it.

### PR DESCRIPTION
**Pull request by rexim**
#502 is mine now as per https://github.com/yegor256/cactoos/issues/502#issuecomment-360621144 and #503 seems to be dead at this point.

Based on #503. Fixed the remaining remarks: formating + more tests.

Close #502, close #503

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#605](https://github.com/yegor256/cactoos/issues/605) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#605](https://github.com/yegor256/cactoos/issues/605) is assigned to @neonailol[/z](https://www.0crat.com/u/neonailol). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

File: src/test/java/org/cactoos/io/GzipOutputTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            "Can't write to a gzip output",
+            baos.toByteArray(),
+            Matchers.equalTo(bytes)
+        );
+    }
+
+    @Test(expected = IOException.class)
+    public void writeToClosedGzipOutput() throws Exception {
+        final OutputStream stream = new FileOutputStream(
+            Files.createTempFile("cactoos", "txt").toFile()
```
Comment (neonailol) [inline src/test/java/org/cactoos/io/GzipOutputTest.java]: @rexim can you update this part to use org.cactoos.io.TempFile?

File: src/main/java/org/cactoos/io/GzipInput.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.GZIPInputStream;
+import org.cactoos.Input;
+
+/**
+ * Input that reads from a Gzip content.
```
Comment (neonailol) [inline src/main/java/org/cactoos/io/GzipInput.java]: please correct spelling to smth like `input that reads compressed data from the GZIP file format` to sound more english

File: src/main/java/org/cactoos/io/GzipOutput.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+package org.cactoos.io;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.zip.GZIPOutputStream;
+import org.cactoos.Output;
+
+/**
+ * Output that writes to a Gzip content.
```
Comment (neonailol) [inline src/main/java/org/cactoos/io/GzipOutput.java]: please correct spelling to smth like `Output that writes compressed data in the GZIP file format.` to sound more english

**Comment by rexim** [comment]
@neonailol done. Please take a look again when you get a chance.

**Comment by neonailol** [comment]
@rultor good to merge

**Comment by neonailol** [comment]
@yegor256 please take a look at this

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Order was successfully finished: +15 points just awarded to @neonailol[/z](https://www.0crat.com/u/neonailol), total is [+525](https://www.0crat.com/u/neonailol)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://datum.zerocracy.com/pages/policy.html#28): +15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+9595](https://www.0crat.com/u/yegor256)

**Comment by 0crat** [comment]
The job [#605](https://github.com/yegor256/cactoos/issues/605) is now out of scope

**Pull request merged**